app/_types/station.ts

```diff
@@ -0,0 +1,33 @@
+export type StationForm = {
+  stationId?: string;
+  ships?: {
+    value: string;
+  }[];
+  modules?: {
+    value: string;
+  }[];
+  commodities?: {
+    value: string;
+  }[];
+  system?: string;
+  minorFaction?: string;
+  allegiance?: string;
+  government?: string;
+  landingPadSize?: string;
+  maxDistanceToArrival?: string;
+  facilities?: {
+    value: string;
+  }[];
+  stationType?: string;
+  requiresPermit?: boolean;
+  power?: string;
+  powerEffect?: string;
+  factionState?: string;
+  economy?: string;
+  nearestSystem?: string;
```

Comment:
Unless I'm misunderstanding, these are all optional fields. Even the station itself.

Purely modelled on old EDDB logic, you could fill out the station name (stationId here) and search, or you could fill out the filters (ie the rest of the fields) and find stations that match - or, iirc, you could fill nothing in, hit search, and it would return something. I think it might have been stations in Sol?

So really this comes down to my point about this being a first draft, EDDB modelled version of the page - there's probably some discussion to be had about how we want our forms to work. I personally don't like the whole convoluted, single form where you can search for a specific station _or_ search by filters, but this is how EDDB did it

Reply:
I see what you're saying, thanks for clarifying. It's been quite a while since I've used EDDB or played ED, so some things I'll need caught up on. It's too bad there's so many optional props, but if this is what gets the template off the ground then so be it.

At another time, I'd like to hear more about your ideas to separate some of these forms. Perhaps creating a guided UX flow would fit the bill.